web/v1/snowflake/sql-runner/sql/standard/00-setup/01-main/01-stored-procedures.sql

```diff
@@ -86,74 +86,99 @@ CREATE OR REPLACE PROCEDURE {{.output_schema}}.column_check(SOURCE_SCHEMA VARCHA
-  var sourceColumns = list_cols_with_type(SOURCE_SCHEMA,SOURCE_TABLE,delim).split(delim);
-  var targetColumns = list_cols_with_type(TARGET_SCHEMA,TARGET_TABLE,delim).split(delim);
-
-  if (targetColumns.some(notIncludedIn(sourceColumns)) === true) {
-
-      throw "ERROR: Source table is missing column(s) which exist in target table.";
-
-  } else {
-
-      var columnAdditions = sourceColumns.filter(notIncludedIn(targetColumns));
-
-      if (sourceColumns.some(notIncludedIn(targetColumns)) === true) {
-          if ( AUTOMIGRATE !== 'TRUE' ) {
-              throw "ERROR: Target table is missing column(s),but automigrate is not enabled.";
-          } else {
-              // enforce order
-              for (var i = 0; i < targetColumns.length; i++) {
-                  if (targetColumns[i] !== sourceColumns[i]) {
-                      throw "ERROR: Can only migrate extra columns of the end of source"
-                  }
-              }
-              add_columns_to(TARGET_SCHEMA, TARGET_TABLE, columnAdditions.join(','));
-              return "ok.Columns added."
-          }
-      } else {
-          return "ok. Columns match."
-      }
+  column_check_stmt = `
+    WITH source_data AS (
+      SELECT
+        column_name,
+        ordinal_position,
+        character_maximum_length,
+        data_type,
+        CASE
+          WHEN isc.data_type='TEXT' THEN CONCAT(isc.column_name, ' VARCHAR(',isc.character_maximum_length, ')')
+          WHEN isc.data_type='NUMBER' THEN CONCAT(isc.column_name, ' NUMBER(', isc.numeric_precision, ',',isc.numeric_scale, ')')
+          else CONCAT(isc.column_name, ' ', isc.data_type)
+        END AS column_definition
+      FROM information_schema.columns isc
+      WHERE table_schema = UPPER(:1)
+      AND table_name = UPPER(:2)
+      ORDER BY ordinal_position
+      ),
+
+  target_data AS (
+    SELECT
+      column_name,
+      ordinal_position,
+      character_maximum_length,
+      data_type
+    FROM information_schema.columns isc
+    where table_schema = UPPER(:3)
+    AND table_name = UPPER(:4)
+    ORDER BY ordinal_position
+    ),
+
+  varchar_check AS (
+    SELECT listagg(
+          CASE WHEN T.character_maximum_length < s.character_maximum_length THEN s.column_name END
+          , ', ') WITHIN GROUP (ORDER BY s.ordinal_position) AS cols_with_varchar_issue
+    FROM target_data  t
+    LEFT JOIN SOURCE_data s
+    ON t.ordinal_position = s.ordinal_position AND t.column_name = s.column_name
+    WHERE t.ordinal_position>= s.ordinal_position),
+
+  type_check AS (
+    SELECT SUM(CASE WHEN s.column_name IS NULL THEN 1 ELSE 0 END) AS missing_in_source,
+          SUM(CASE WHEN t.column_name IS NULL THEN 1 ELSE 0 END) AS missing_in_target
+    FROM target_data  t
+    FULL OUTER JOIN SOURCE_data s
+    ON t.ordinal_position = s.ordinal_position AND T.data_type = s.data_type AND s.column_name = t.column_name
+    WHERE t.ordinal_position>= s.ordinal_position OR s.ordinal_position is null),
```

Comment:
Again slightly confused by this filter. What is its purpose?

Reply:
Thank you for highlighting this. The code has been revised in a later commit so this became redundant.